### ISSUE TITLE
Crashes when returning recently created node. (2.0-edge)

### ISSUE THREAD

**Issue by gnaneshr** [RedisGraph/RedisGraph#502]
In the tree: `9e29db1bd72ff2ff4b5eabe2ce5cd0fe5b0fe155` which has an alpha implementation of RedisGraph 2.0, there is a problem in returning the recently created node.

Server:
`sudo docker run -p 6379:6379 -it --rm redislabs/redisgraph:2.0-edge`

Client:
```
127.0.0.1:6379> GRAPH.QUERY ll "CREATE (:Company { name: 'Redis Labs' })"
1) 1) "Labels added: 1"
   2) "Nodes created: 1"
   3) "Properties set: 1"
   4) "Query internal execution time: 1.444328 milliseconds"
127.0.0.1:6379> GRAPH.QUERY ll "MATCH (c { name: 'Redis Labs' }) CREATE (e:Employee { name: 'Roi Lipman' })-[:WORK]->(c) RETURN e, c"
Error: Connection reset by peer
127.0.0.1:6379> 
```

This assertion was logged by the server:
```
redis-server: /redisgraph/src/resultset/resultset_replyverbose.c:34: _ResultSet_VerboseReplyWithSIValue: Assertion `"Unhandled value type" && false' failed.
```

Here's the bug report:
```
=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 28 May 2019 05:16:48.936 # Redis 5.0.5 crashed by signal: 11
1:M 28 May 2019 05:16:48.936 # Crashed running the instruction at: 0x7f6466d98529
1:M 28 May 2019 05:16:48.936 # Accessing address: (nil)
1:M 28 May 2019 05:16:48.936 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
/lib/x86_64-linux-gnu/libc.so.6(abort+0x269)[0x7f6466d98529]

Backtrace:
redis-server *:6379(logStackTrace+0x37)[0x5593fe9413e7]
redis-server *:6379(sigsegvHandler+0xb0)[0x5593fe941b10]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x110e0)[0x7f64671140e0]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x269)[0x7f6466d98529]
/lib/x86_64-linux-gnu/libc.so.6(+0x2be67)[0x7f6466d8fe67]
/lib/x86_64-linux-gnu/libc.so.6(+0x2bf12)[0x7f6466d8ff12]
/usr/lib/redis/modules/redisgraph.so(+0x820bf)[0x7f646494e0bf]
/usr/lib/redis/modules/redisgraph.so(ResultSet_EmitVerboseRecord+0x41b)[0x7f646494e55b]
/usr/lib/redis/modules/redisgraph.so(ResultSet_AddRecord+0x20)[0x7f646494d890]
/usr/lib/redis/modules/redisgraph.so(ResultsConsume+0x31)[0x7f646493c781]
/usr/lib/redis/modules/redisgraph.so(ExecutionPlan_Execute+0x26)[0x7f6464936786]
/usr/lib/redis/modules/redisgraph.so(_MGraph_Query+0x360)[0x7f6464933ec0]
/usr/lib/redis/modules/redisgraph.so(+0x85ba4)[0x7f6464951ba4]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x74a4)[0x7f646710a4a4]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f)[0x7f6466e4cd0f]

------ INFO OUTPUT ------
# Server
redis_version:5.0.5
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:7983a619928f1f2d
redis_mode:standalone
os:Linux 4.15.0-50-generic x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:6.3.0
process_id:1
run_id:63c7ea63e12effbc630b987633360749ddc52fb6
tcp_port:6379
uptime_in_seconds:77
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:15516736
executable:/data/redis-server
config_file:

# Clients
connected_clients:1
client_recent_max_input_buffer:2
client_recent_max_output_buffer:0
blocked_clients:1

# Memory
used_memory:1841264
used_memory_human:1.76M
used_memory_rss:6000640
used_memory_rss_human:5.72M
used_memory_peak:1841264
used_memory_peak_human:1.76M
used_memory_peak_perc:103.69%
used_memory_overhead:841126
used_memory_startup:791360
used_memory_dataset:1000138
used_memory_dataset_perc:95.26%
allocator_allocated:2090320
allocator_active:2363392
allocator_resident:5255168
total_system_memory:16711352320
total_system_memory_human:15.56G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.13
allocator_frag_bytes:273072
allocator_rss_ratio:2.22
allocator_rss_bytes:2891776
rss_overhead_ratio:1.14
rss_overhead_bytes:745472
mem_fragmentation_ratio:3.46
mem_fragmentation_bytes:4265896
mem_not_counted_for_evict:0
mem_replication_backlog:0
mem_clients_slaves:0
mem_clients_normal:49694
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0

# Persistence
loading:0
rdb_changes_since_last_save:2
rdb_bgsave_in_progress:0
rdb_last_save_time:1559020531
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0

# Stats
total_connections_received:1
total_commands_processed:2
instantaneous_ops_per_sec:0
total_net_input_bytes:215
total_net_output_bytes:136
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
evicted_keys:0
keyspace_hits:1
keyspace_misses:1
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0

# Replication
role:master
connected_slaves:0
master_replid:b8d0196e2fcf8fb61f4f30d2c47b151f00209632
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.125350
used_cpu_user:0.175490
used_cpu_sys_children:0.000000
used_cpu_user_children:0.003104

# Commandstats
cmdstat_graph.QUERY:calls=2,usec=369,usec_per_call=184.50

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=1,expires=0,avg_ttl=0

------ CLIENT LIST OUTPUT ------
id=3 addr=172.17.0.1:47994 fd=8 name= age=38 idle=0 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=0 oll=0 omem=0 events=r cmd=graph.QUERY

------ REGISTERS ------
1:M 28 May 2019 05:16:48.949 # 
RAX:0000000000000006 RBX:00007f6467c52000
RCX:00007f6466d96fff RDX:0000000000000000
RDI:0000000000000002 RSI:00007f64640c8160
RBP:00007f64649bdb58 RSP:00007f64640c8270
R8 :0000000000000000 R9 :00007f64640c8160
R10:0000000000000008 R11:0000000000000246
R12:0000000000000022 R13:00007f64649bdc20
R14:0000000000000000 R15:0000000000000002
RIP:00007f6466d98529 EFL:0000000000010246
CSGSFS:002b000000000033
1:M 28 May 2019 05:16:48.950 # (00007f64640c827f) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c827e) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c827d) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c827c) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c827b) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c827a) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8279) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8278) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8277) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8276) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8275) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8274) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8273) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8272) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8271) -> 0000000000000000
1:M 28 May 2019 05:16:48.950 # (00007f64640c8270) -> 0000000000000020

------ FAST MEMORY TEST ------
1:M 28 May 2019 05:16:48.950 # Bio thread for job type #0 terminated
1:M 28 May 2019 05:16:48.951 # Bio thread for job type #1 terminated
1:M 28 May 2019 05:16:48.951 # Bio thread for job type #2 terminated
*** Preparing to test memory region 5593fec77000 (2252800 bytes)
*** Preparing to test memory region 5593ff9f8000 (237568 bytes)
*** Preparing to test memory region 7f6454000000 (135168 bytes)
*** Preparing to test memory region 7f645c000000 (143360 bytes)
*** Preparing to test memory region 7f6460000000 (8388608 bytes)
*** Preparing to test memory region 7f64608c5000 (8388608 bytes)
*** Preparing to test memory region 7f64610c6000 (8388608 bytes)
*** Preparing to test memory region 7f64618c7000 (8388608 bytes)
*** Preparing to test memory region 7f64620c8000 (8388608 bytes)
*** Preparing to test memory region 7f64628c9000 (8388608 bytes)
*** Preparing to test memory region 7f64630ca000 (8388608 bytes)
*** Preparing to test memory region 7f64638cb000 (8388608 bytes)
*** Preparing to test memory region 7f64640cc000 (8388608 bytes)
*** Preparing to test memory region 7f6464bfc000 (4096 bytes)
*** Preparing to test memory region 7f6464bfe000 (8388608 bytes)
*** Preparing to test memory region 7f64653ff000 (8388608 bytes)
*** Preparing to test memory region 7f6465c00000 (8388608 bytes)
*** Preparing to test memory region 7f6466400000 (8388608 bytes)
*** Preparing to test memory region 7f64670ff000 (16384 bytes)
*** Preparing to test memory region 7f646731c000 (16384 bytes)
*** Preparing to test memory region 7f6467bc5000 (409600 bytes)
*** Preparing to test memory region 7f6467c49000 (32768 bytes)
*** Preparing to test memory region 7f6467c52000 (4096 bytes)
*** Preparing to test memory region 7f6467c55000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.
```

And here's the backtrace when compiled with RedisGraph(`9e29db1bd72ff2ff4b5eabe2ce5cd0fe5b0fe155`) and redis (latest version)with `-g3` flag):
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007f165c994801 in __GI_abort () at abort.c:79
#2  0x00007f165c98439a in __assert_fail_base (fmt=0x7f165cb0b7d8 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x7f16597315d0 "\"Unhandled value type\" && false", 
    file=file@entry=0x7f1659731588 "/home/gnanesh_r/git/RedisGraph/src/resultset/resultset_replyverbose.c", line=line@entry=34, 
    function=function@entry=0x7f1659731620 <__PRETTY_FUNCTION__.12082> "_ResultSet_VerboseReplyWithSIValue") at assert.c:92
#3  0x00007f165c984412 in __GI___assert_fail (assertion=0x7f16597315d0 "\"Unhandled value type\" && false", 
    file=0x7f1659731588 "/home/gnanesh_r/git/RedisGraph/src/resultset/resultset_replyverbose.c", line=34, 
    function=0x7f1659731620 <__PRETTY_FUNCTION__.12082> "_ResultSet_VerboseReplyWithSIValue") at assert.c:101
#4  0x00007f165966d7d8 in _ResultSet_VerboseReplyWithSIValue (ctx=0x7f1653800000, gc=0x7f1653c18000, v=...)
    at /home/gnanesh_r/git/RedisGraph/src/resultset/resultset_replyverbose.c:34
#5  0x00007f165966dda3 in ResultSet_EmitVerboseRecord (ctx=0x7f1653800000, gc=0x7f1653c18000, r=0x7f1653824050, numcols=1)
    at /home/gnanesh_r/git/RedisGraph/src/resultset/resultset_replyverbose.c:145
#6  0x00007f165966a83c in ResultSet_AddRecord (set=0x7f16440011d0, r=0x7f1653824050) at /home/gnanesh_r/git/RedisGraph/src/resultset/resultset.c:170
#7  0x00007f1659624a52 in ResultsConsume (opBase=0x7f16440020a0) at /home/gnanesh_r/git/RedisGraph/src/execution_plan/ops/op_results.c:40
#8  0x00007f16596094c7 in ExecutionPlan_Execute (plan=0x7f1644001a40) at /home/gnanesh_r/git/RedisGraph/src/execution_plan/execution_plan.c:700
#9  0x00007f1659603867 in _MGraph_Query (args=0x7f165c415140) at /home/gnanesh_r/git/RedisGraph/src/commands/cmd_query.c:113
#10 0x00007f165967ff51 in thread_do (thread_p=0x55fce4ba5120) at /home/gnanesh_r/git/RedisGraph/src/util/thpool/thpool.c:337
#11 0x00007f165cd4c6db in start_thread (arg=0x7f1658bf6700) at pthread_create.c:463
#12 0x00007f165ca7588f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by jeffreylovitz** [comment]
@gnaneshr Thanks for the report!

This shouldn't take long to fix, we'll report back here soon.

**Comment by swilly22** [comment]
Hi @gnaneshr,
Please confirm fix #508 merged to master branch.

**Comment by gnaneshr** [comment]
@swilly22 It's fixed.
